gr-azure-software-radio/python/eventhub_sink.py

```diff
@@ -0,0 +1,127 @@
+    Event Hub Name: The Event Hub must be created in Azure before trying to
+        send events to it.
+    Partition ID: The partition id to send events to.
+    """
+    # pylint: disable=too-many-arguments, no-member
+    def __init__(
+            self,
+            authentication_method: str = "default",
+            connection_str: str = None,
+            sas_token: str = None,
+            eventhub_host_name: str = None,
+            eventhub_name: str = None,
+            partition_id: str = None):
+
+        gr.sync_block.__init__(self,
+                               name="eventhub_sink",
+                               in_sig=[],
+                               out_sig=[])
+
+        self.partition_id = partition_id
+
+        self.eventhub_producer = get_eventhub_producer_client(
+            authentication_method=authentication_method,
+            eventhub_name=eventhub_name,
+            connection_str=connection_str,
+            sas_token=sas_token,
+            eventhub_host_name=eventhub_host_name
+        )
+
+        self.message_port_register_in(pmt.intern('in'))
+        self.set_msg_handler(pmt.intern('in'), self.handle_msg)
+
+    def handle_msg(self, msg):
+        """
+        This function handles the message passing operating (PMT message) from GNU Radio.
+        It converts the PMT structure to python and later to JSON to send as an event.
+        """
+        pmsg = pmt.to_python(msg)
+        jmsg = json.dumps(pmsg)
+        event_batch = self.eventhub_producer.create_batch(
+            partition_id=self.partition_id)
+        event_batch.add(EventData(jmsg))
+        self.eventhub_producer.send_batch(event_batch)
+
+    def stop(self):
+        self.eventhub_producer.close()
+        return True
+
+
+def get_eventhub_producer_client(
+        authentication_method: str = "default",
+        connection_str: str = None,
+        sas_token: str = None,
+        eventhub_host_name: str = None,
+        eventhub_name: str = None):
+    """ Initialize the Event Hub Producer client
+
+    Args:
+        authentication_method (str): Determines which method to use to authenticate to Eventhub.
+         May be one of "connection_string", "sas", or "default".
+        connection_str (optional, str): Eventhub connection string used for
+            authentication if authentication_method is "connection_string"
+        sas_token (optional, str): Shared Access Signature Token. This is required if using
+            "sas" authentication.
+        eventhub_host_name (optional, str): The fully qualified host name for the Event Hub namespace. This
+            is required if using "sas" and "default" authentication.
+        eventhub_name (str): The path to the specified Event Hub to connect to.
+    Raises:
+        ValueError: Raised if an unsupported authentication method is used
+
+    Returns:
+        EventHubProducerClient: An Event Hub producer client ready to be used
+    """
```

Comment:
Shouldn't this integrate with Marko's default creds stuff?

Reply:
I created user story ADO#8095 to address this. The Blob blocks can also benefit from the DefaultCredentials block.